File: cases/taint/scala/return-relay-one-hop-negative/ReturnRelayOneHopNegative.scala
package dataflowbench

object ReturnRelayOneHopNegative {
  def dfb_source(): Int = { // DFB-SOURCE: return-one-hop-negative-input
    1
  }

  def relay(value: Int): Int = { // DFB-WITNESS: return-one-hop-negative-relay
    value
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: return-one-hop-negative-sink

  def run(): Unit = {
    val result = relay(dfb_source())
    dfb_sink(0)
  }
}

File: cases/taint/rust/return-relay-one-hop-negative/return_relay_one_hop_negative.rs
fn dfb_source() -> i32 { // DFB-SOURCE: return-one-hop-negative-input
    1
}

fn relay(value: i32) -> i32 { // DFB-WITNESS: return-one-hop-negative-relay
    value
}

fn dfb_sink(value: i32) {} // DFB-SINK: return-one-hop-negative-sink

fn run() {
    let result = relay(dfb_source());
    dfb_sink(0);
}
